app/bundles/EmailBundle/Tests/Swiftmailer/SendGrid/Mail/SendGridMailMetadataTest.php

```diff
@@ -12,17 +12,41 @@
 namespace Mautic\EmailBundle\Tests\Swiftmailer\SendGrid\Mail;
 
 use Mautic\EmailBundle\Swiftmailer\SendGrid\Mail\SendGridMailMetadata;
+use PHPUnit\Framework\MockObject\MockObject;
 use SendGrid\BccSettings;
 use SendGrid\Mail;
 use SendGrid\MailSettings;
 use SendGrid\ReplyTo;
 
 class SendGridMailMetadataTest extends \PHPUnit\Framework\TestCase
 {
+    private function make_header(string $key, string $value): MockObject
+    {
+        $an_header = $this->getMockBuilder(\Swift_Mime_Header::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $an_header->method('getFieldName')
+            ->willReturn($key);
+
+        $an_header->method('getFieldBody')
+            ->willReturn($value);
+
+        return $an_header;
+    }
+
     public function testBaseMessage()
     {
         $sendGridMailMetadata = new SendGridMailMetadata();
 
+        $headers = $this->getMockBuilder(\Swift_Mime_SimpleHeaderSet::class)
+            ->disableOriginalConstructor()
+            ->getMock();
```

Comment:
Better readability with
```suggestion
        $headers = $this->createMock(\Swift_Mime_SimpleHeaderSet::class);
```